apps/framework-cli/src/infrastructure/stream/redpanda.rs

```diff
@@ -48,15 +48,9 @@ pub async fn execute_changes(
             }
 
             StreamingChange::Topic(Change::Updated { before, after }) => {
-                if !project.is_production {
-                    info!("Replacing topic: {:?} with: {:?}", before, after);
-                    delete_topics(&project.redpanda_config, vec![before.id()]).await?;
-                    create_topics(&project.redpanda_config, vec![after.id()]).await?;
-                } else {
-                    return Err(RedpandaChangesError::NotSupported(format!(
-                        "Updating topic {} is not supported in production mode",
-                        before.id()
-                    )));
+                if before.retention_period != after.retention_period {
```

Comment:
i worry that we might add another config to topic, but forgot to add the condition to this check

Reply:
Cool - will refactor with a diff on the field called in the topic

Reply:
Actually, while working on the code, I also noticed that the update_topic_config is very specific to Kafka and would need to be updated too. 
I don't think we should be moving that implementation close to the `Topic` implementation since it is specify to Kafka/Redpanda, and thus should stay in this file.

Do you have a suggestion of a pattern that could help achieve this?